src/lib/bootloader/serial_console.rb

```diff
@@ -0,0 +1,70 @@
+require "yast"
+
+Yast.import "Arch"
+
+module Bootloader
+  class SerialConsole
+    PARITY_MAP = {
+      "n" => "no",
+      "o" => "odd",
+      "e" => "even"
+    }
+    SPEED_DEFAULT = 9600
+    PARITY_DEFAULT = "no"
+    WORD_DEFAULT = ""
+
+    def self.load_from_kernel_args(kernel_params)
+      console_parameter = kernel_params.parameter("console")
+      return nil unless console_parameter
+
+      console_parameter = [console_parameter] unless console_parameter.is_a? ::Array
+      serial_console = console_parameter.find { |p| p =~ /ttyS/ || p =~ /ttyAMA/ }
+      return nil unless serial_console
+
+      console_regexp = /(ttyS|ttyAMA)[[:alpha:]]+([[:digit:]]*),?([[:digit:]]*)([noe]*)([[:digit:]]*)/
+      unit = serial_console[console_regexp, 2]
+      return nil if unit.empty
+
+      speed = serial_console[console_regexp, 3]
+      speed = SPEED_DEFAULT if speed.empty?
+      parity = serial_console[console_regexp, 4]
+      parity = PARITY_DEFAULT if parity.empty?
+      parity = PARITY_MAP[parity]
+      word = serial_console[console_regexp, 5]
+
+      new(unit, speed, parity, word)
+    end
+
+    def self.load_from_console_args(console_args)
```

Comment:
Ok, I know that documentation is missing yet, but some explanation about `console_args` will help :)